### PR TITLE
Add InitialMmapSize to bolt options

### PR DESCRIPTION
**Pull request by raskchanky**
https://hashicorp.atlassian.net/browse/VAULT-2552

Since we're rapidly accumulating options that we want to share between raft.db and vault.db, I refactored the `freelistOptions()` function to be more generally `boltOptions()`. I'm using `strconv.IntSize` to determine whether we're on a 32bit or 64bit architecture, since we only want the InitialMmapSize to be set for 64bit platforms. I'm including a few screenshots here of a simple program I wrote to test that this works the way I think it should.

This is the program in its entirety, running locally on my MBP.
CleanShot 2021-11-16 at 09 26 26

This is the same program, running on 32bit Debian Linux, in a VM.
CleanShot 2021-11-16 at 09 29 26

In addition to the program and its output, I also included the output of a few different utilities, in an effort to demonstrate that this VM is really running 32bit Linux.

I also included a few tests, one 32 bit specific, to make sure that we're not setting InitialMmapSize on 32bit platforms.

### CODE REVIEW

**Comment by ncabatoff** [comment]
I don't understand the package build failures, but it's suspicious that all the failing ones are 32-bit targets.

**Comment by raskchanky** [comment]
@ncabatoff I tried compiling this on my 32bit VM and it errored on this line:
`o.InitialMmapSize = 100 * 1024 * 1024 * 1024`
saying that it overflows `int`.

So it appears that I need to re-work this to use platform specific build tags instead.

**Comment by raskchanky** [comment]
I'm also not sure if this requires a changelog or not, since it's not a user visible change. Thoughts?

**Comment by ncabatoff** [comment]
> I'm also not sure if this requires a changelog or not, since it's not a user visible change. Thoughts?

My position is that if a change affects the compiled binary, it deserves a changelog.

File: physical/raft/raft.go [hunk 1]
```diff
@@ -1644,20 +1644,38 @@ func (s sealer) Open(ctx context.Context, ct []byte) ([]byte, error) {
...
+	// Otherwise, we set it to whatever the value of VAULT_RAFT_INITIAL_MMAP_SIZE
+	// is, assuming it can be parsed as an int. Bolt itself sets this to 0 by default,
+	// so if users are wanting to turn this off, they can also set it to 0. Setting it
+	// to a negative value is the same as not setting it at all.
+	if os.Getenv("VAULT_RAFT_INITIAL_MMAP_SIZE") == "" {
+		o.InitialMmapSize = initialMmapSize
+	} else {
+		imms, err := strconv.Atoi(os.Getenv("VAULT_RAFT_INITIAL_MMAP_SIZE"))
+		if err != nil {
+			return nil, err
```
Comment (ncabatoff) [inline physical/raft/raft.go:1674]: Are you sure you want to fail startup if someone puts a non-numeric value?  I'm ok with it personally, but it seems a little inconsistent with how we handle these feature flags normally, e.g. VAULT_RAFT_FREELIST_TYPE=bogus doesn't fail startup.

File: physical/raft/vars_64bit.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// +build amd64 arm64 s390x
```
[the remark this answers is not in the document]
Comment (ncabatoff) [reply physical/raft/vars_64bit.go:1]: Also, why not use the same build pragmas for the test files as for these files?

**Review by ncabatoff — APPROVED** [review]

File: physical/raft/raft.go (hunk 1 above)
Comment (raskchanky) [reply physical/raft/raft.go:1674]: I think I default to wanting to let the user know if they did something wrong, but I can change it to just ignore it.

**Review by ncabatoff — COMMENTED** [review]
LGTM but you have failing tests

**Pull request merged**